lib/fs.ts

```diff
@@ -178,7 +178,7 @@ class FSPath {
 			throttleTimer: null,
 		};
 		pendingUpdates.set(this.path, update);
-		this.safeWrite(dataFetcher(), options).then(() => this.finishUpdate());
+		return this.safeWrite(dataFetcher(), options).then(() => this.finishUpdate());
```

Comment:
Now here's the question – do you actually want to return this Promise? It's also meaningless – under normal conditions, this Promise will never resolve; it's an infinite loop repeatedly writing the latest update.

Reply:
But sometimes it does resolve, if nothing needs to be updated for a while. Nothing good would happen if someone tried to await this Promise.

Reply:
Similar concerns apply to the next two functions, by the way. The Promises you're newly returning are meaningless – all the meaningful Promises already being used.